### PR TITLE
`json`, `ipaddress`, `locale`: Python3.13 updates

### PR DESCRIPTION
**Pull request by aminalaee**
- Add `ipaddress.IPv4Address.ipv6_mapped`
- Remove `locale.resetlocale` in Py 3.13
- Update `json.encoder.encode_basestring` `json.encoder.encode_basestring_ascii`

### CODE REVIEW

File: stdlib/ipaddress.pyi [hunk 1]
```diff
@@ -146,6 +146,8 @@ class _BaseV4:
     def version(self) -> Literal[4]: ...
     @property
     def max_prefixlen(self) -> Literal[32]: ...
+    @property
+    def ipv6_mapped(self) -> IPv6Address: ...
```
Comment (AlexWaygood) [inline stdlib/ipaddress.pyi:150]:
```suggestion
    if sys.version_info >= (3, 13):
        @property
        def ipv6_mapped(self) -> IPv6Address: ...
```

**Comment by aminalaee** [comment]
@AlexWaygood I couldn't find any examples for this, is there a way to annotate future types in typeshed?
The `IPV6Address` is defined after the `IPV4Address` but this property needs to return that.

**Comment by AlexWaygood** [comment]
> @AlexWaygood I couldn't find any examples for this, is there a way to annotate future types in typeshed? The `IPV6Address` is defined after the `IPV4Address` but this property needs to return that. Would `"IPV6Address"` work?

Forward references are natively supported in stub files; it's as if `from __future__ import annotations` was imported at the top of every stub file. Just returning `IPV6Address`, without the quotes, should be fine :-)

**Comment by AlexWaygood** [comment]
> Yeah you're right, the issue seems to be that the `IPV4Address. ipv6_mapped ` is not available in Python3.13 (yet). I faced a similar case with the `glob.translate` that it exists in the 3.13 branch but not in the release.

No, it is available on 3.13.0b1. But it's implemented at runtime on `ipaddress.IPV4Address`, not `ipaddress._BaseV4`. Your PR currently implements it on `ipaddress._BaseV4`, which isn't quite right, and is why stubtest is complaining :-)

**Comment by AlexWaygood** [comment]
(`glob.translate()` should also be included as part of Python 3.13.0b1, FWIW :-)

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Pull request merged**